src/tools/kata-ctl/src/monitor/http_server.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022-2023 Kata Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::monitor::metrics::get_monitor_metrics;
+use crate::sl;
+use anyhow::{anyhow, Context, Result};
+use hyper::body;
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use shim_interface::shim_mgmt::client::MgmtClient;
+use slog::{self, info};
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::time::Duration;
+
+const ROOT_URI: &str = "/";
+const METRICS_URI: &str = "/metrics";
+
+const MONITOR_METRICS_CTX: &str = "failed to get monitor metrics";
+const RUNTIME_METRICS_CTX: &str = "failed to get runtime metrics";
+
+async fn handler_mux(req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
+    info!(
+        sl!(),
+        "mgmt-svr(mux): recv req, method: {}, uri: {}",
+        req.method(),
+        req.uri().path()
+    );
+
+    match (req.method(), req.uri().path()) {
+        (&Method::GET, ROOT_URI) => root_uri_handler(req).await,
+        (&Method::GET, METRICS_URI) => metrics_uri_handler(req).await,
+        _ => Ok(not_found_uri_handler(req).await),
+    }
+}
+
+#[tokio::main]
+pub async fn http_server_setup(socket_addr: &str) -> Result<()> {
+    let addr: SocketAddr = socket_addr
+        .parse()
+        .context("failed to parse http socket address")?;
+
+    let make_svc =
+        make_service_fn(|_conn| async { Ok::<_, hyper::Error>(service_fn(handler_mux)) });
+
+    if let Err(e) = Server::bind(&addr).serve(make_svc).await {
+        eprintln!("server error: {}", e);
+    }
+
+    Ok(())
+}
+
+async fn root_uri_handler(_req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
+    Ok(Response::builder()
+        .status(StatusCode::OK)
+        .body(Body::from(
+            r#"Available HTTP endpoints:
+    /metrics : Get metrics from sandboxes.
+"#,
+        ))
+        .unwrap())
+}
+
+async fn metrics_uri_handler(req: Request<Body>) -> Result<Response<Body>, hyper::Error> {
+    let mut response_body = String::new();
+
+    response_body += &get_monitor_metrics().unwrap_or_else(|e| {
+        eprintln!("ERROR: {} {:?}", MONITOR_METRICS_CTX, e);
```

Comment:
This function returns a `Result` so why not just return the error and let the caller handle it?